### PR TITLE
Add Garden resource metrics to the runtime kube-state-metrics

### PR DESCRIPTION
**Pull request by rickardsjp**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:
This PR introduces metrics for the `Garden` resource in the garden runtime cluster. This enables monitoring of the conditions and last operation.
To achieve this, it builds on previous enhancements to the kube-state-metrics deployment, notably bebad97 and 1e3de9d, which enabled monitoring the state of a custom resource using kube-state-metrics. Read more about custom resource state metrics [here](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/extend/customresourcestate-metrics.md).

The resulting metrics will look something like the following. For readability's sake, I have omitted all but one condition and also removed the last operation metric in the below sample:

```
# HELP garden_garden_condition represents a condition of a Garden object
# TYPE garden_garden_condition stateset
garden_garden_condition{condition="ObservabilityComponentsHealthy",customresource_group="operator.gardener.cloud",customresource_kind="Garden",customresource_version="v1alpha1",name="garden",status="False"} 0
garden_garden_condition{condition="ObservabilityComponentsHealthy",customresource_group="operator.gardener.cloud",customresource_kind="Garden",customresource_version="v1alpha1",name="garden",status="Progressing"} 0
garden_garden_condition{condition="ObservabilityComponentsHealthy",customresource_group="operator.gardener.cloud",customresource_kind="Garden",customresource_version="v1alpha1",name="garden",status="True"} 1
garden_garden_condition{condition="ObservabilityComponentsHealthy",customresource_group="operator.gardener.cloud",customresource_kind="Garden",customresource_version="v1alpha1",name="garden",status="Unknown"} 0
```

For each condition, the StateSet will enumerate the possible states as values for the status label, and identify the currently active condition by its sample value. The last operation metric works similarly: It enumerates all operation types and identifies the *actual* last operation using the sample value.
This is a viable approach because there is a limited number of possible values for the labels in question.

**Special notes for your reviewer**:
/cc @istvanballok @vicwicker @chrkl 

**Release note**:

```feature operator
Introduced metrics exposing the Garden resource's condition and last operation.
```

### CODE REVIEW

File: pkg/component/observability/monitoring/kubestatemetrics/resources.go [hunk 1]
```diff
@@ -391,6 +391,8 @@ var gardenMetricAllowlist = []string{
 	"^kube_customresource_verticalpodautoscaler_spec_resourcepolicy_containerpolicies_maxallowed_cpu$",
 	"^kube_customresource_verticalpodautoscaler_spec_resourcepolicy_containerpolicies_maxallowed_memory$",
 	"^kube_customresource_verticalpodautoscaler_spec_updatepolicy_updatemode$",
+	"^garden_garden_condition$",
```
Comment (chrkl) [inline pkg/component/observability/monitoring/kubestatemetrics/resources.go:399]: I think the `garden_` prefix has to be replaced by `kube_customresource_` here (see VPA metrics above).

Comment (chrkl) [reply pkg/component/observability/monitoring/kubestatemetrics/resources.go:399]: I was wrong. The name prefix is part of the configuration. An incomplete cluster role was the reason for the metrics not being exported.

Comment (rickardsjp) [reply pkg/component/observability/monitoring/kubestatemetrics/resources.go:399]: You were faster to post this comment than my `make generate` command. I added a commit, the metrics are created and also scraped successfully by the `prometheus-garden` pods.

File: pkg/component/observability/monitoring/kubestatemetrics/customresourcestate_test.go [hunk 2]
```diff
@@ -19,17 +19,26 @@ import (
...
+func expectedCustomResourceStateConfig(suffix string) string {
 	defer GinkgoRecover()
+	var rawActual []byte
+	var expectFilePath string
+	var err error
 
-	rawActual, err := yaml.Marshal(NewCustomResourceStateConfig())
-	Expect(err).ToNot(HaveOccurred())
+	options := []Option{WithVPAMetrics}
+	relativePath := "testdata/custom-resource-state-vpa.expectation.yaml"
```
Comment (rfranzke) [inline pkg/component/observability/monitoring/kubestatemetrics/customresourcestate_test.go]: Nit: Use `var ()` block

File: pkg/component/observability/monitoring/kubestatemetrics/resources.go [hunk 3]
```diff
@@ -714,7 +721,18 @@ func (k *kubeStateMetrics) nameSuffix() string {
...
+
+	suffix := k.values.NameSuffix
+
+	switch suffix {
+	case SuffixRuntime:
+		customResourceStateConfig, err = yaml.Marshal(NewCustomResourceStateConfig(WithVPAMetrics, WithGardenResourceMetrics))
+	default:
+		customResourceStateConfig, err = yaml.Marshal(NewCustomResourceStateConfig(WithVPAMetrics))
+	}
+
```
Comment (rfranzke) [inline pkg/component/observability/monitoring/kubestatemetrics/resources.go]:
```suggestion
	opts := []kubestatemetrics.Option{WithVPAMetrics}
	if k.values.NameSuffix == SuffixRuntime {
		opts = append(opts, WithGardenResourceMetrics)
	}
	
	customResourceStateConfig, err := yaml.Marshal(NewCustomResourceStateConfig(opts...))
```

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**